ludwig/api.py

```diff
@@ -1406,6 +1407,8 @@ def preprocess(
             (proc_training_set, proc_validation_set, proc_test_set, training_set_metadata) = preprocessed_data
 
             return PreprocessedDataset(proc_training_set, proc_validation_set, proc_test_set, training_set_metadata)
+        except Exception as e:
```

Comment:
nit: Add a raises section to the docstring: https://google.github.io/styleguide/pyguide.html#doc-function-raises